File: c#/tema_3/ConcertPro/Dao/Repository/IRepository.cs
using System.Collections.Generic;
using ConcertPro.Dao.Model;

namespace ConcertPro.Dao.Repository
{
    public interface IRepository<T, TId> where T : IEntity<TId>
    {
        T FindOne(TId id);
        IEnumerable<T> FindAll();
        void Save(T entity);
        void Update(T entity, TId id);
        void Delete(TId id);
    }
}

File: java/tema_3/src/main/java/ro/mpp2024/concertpro/dao/repository/IRepository.java
package ro.mpp2024.concertpro.dao.repository;

import ro.mpp2024.concertpro.dao.model.Entity;

public interface IRepository<T extends Entity<ID>, ID> {
    T findOne(ID id);
    Iterable<T> findAll();
    void save(T elem);
    void update(T elem, ID id);
    void delete(ID id);
}
